### PR TITLE
chore: Examples pruning part 1

### PR DESCRIPTION
**Pull request by MikhailKardash**
## Description

Move a large chunk of examples to a legacy directory. Remove their download links and references from docs. Remove their tests from CI. 

## Test Plan

CI passes

## Commentary (optional)

This is part 1 of examples pruning. Only 2 PR-worthy tasks remain: moving tutorials and (re?)moving model_hub v1.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] Docs and github references to the legacy examples have been resolved
- [ ] Blog post references to the legacy examples have been resolved
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket
MLG-869

### CODE REVIEW

**Review by tara-det-ai — COMMENTED** [review]
please edit line 22

original 

For an introduction to using the training API, please visit the Training APIs section.

fix

For an introduction to using the training APIs, please visit :ref:`Training APIs <apis-howto-overview>`.

**Review by tara-det-ai — COMMENTED** [review]
some files are referenced without *.tgz, e.g., `hp_constraints_mnist_pytorch` is referenced in the file `hp-constraints-det.rst`

File: docs/interfaces/tensorboard.rst [hunk 1]
```diff
@@ -159,8 +159,7 @@ PyTorch
 See :func:`PyTorchTrialContext.get_tensorboard_writer()
 <determined.pytorch.PyTorchTrialContext.get_tensorboard_writer>`
 
-For a full-length example of using TensorBoard with PyTorch, check out the :download:`mnist-GAN
-model </examples/gan_mnist_pytorch.tgz>`.
+For a full-length example of using TensorBoard with PyTorch, check out the mnist-GAN model.
```
Comment (tara-det-ai) [inline docs/interfaces/tensorboard.rst:162]: is there a public repo where the user can find this model?

**Review by tara-det-ai — APPROVED** [review]
I've reviewed the *.rst files.

File: docs/interfaces/tensorboard.rst (hunk 1 above)
Comment (MikhailKardash) [reply docs/interfaces/tensorboard.rst:162]: Oh, I think this line should be deleted actually.

File: docs/model-dev-guide/hyperparameter/search-methods/hp-custom.rst [hunk 2]
```diff
@@ -39,8 +39,7 @@ Both search runners execute the custom hyperparameter tuning algorithm and start
 experiment on a Determined cluster.
 
 The following sections explain the steps to take in order to implement and use a custom
-hyperparameter search algorithm. A detailed example can be found in
-:download:`asha_search_method.tgz </examples/asha_search_method.tgz>`.
+hyperparameter search algorithm.
```
Comment (tara-det-ai) [inline docs/model-dev-guide/hyperparameter/search-methods/hp-custom.rst:42]: The following sections describe the steps needed to implement and use a custom
hyperparameter search algorithm.

**Review by tara-det-ai — COMMENTED** [review]
added another suggestion

**Comment by MikhailKardash** [comment]
Oops, was working off of the wrong main. I'll just merge main instead of rebasing next time. Apologies for all of the automated review requests.

**Comment by MikhailKardash** [comment]
Made a new PR for this: https://github.com/determined-ai/determined/pull/7918

**Pull request closed without merging**